apps/api/src/domain/games/memory-match/queries/memory-match-admin.remove-card.e2e.spec.ts

```diff
@@ -0,0 +1,167 @@
+        },
+    };
+
+    const setItUp = async (user?: CoscradUserWithGroups) => {
+        const testModule = await Test.createTestingModule({
+            imports: [
+                ConfigModule.forRoot({
+                    isGlobal: true,
+                    envFilePath: buildConfigFilePath(Environment.test),
+                    cache: false,
+                }),
+                AuthorizationModule,
+                PersistenceModule.forRootAsync(),
+                MemoryMatchModule,
+            ],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService({
+                    ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    NODE_PORT: 3131,
+                })
+            )
+            .overrideGuard(AdminJwtGuard)
+            .useValue(new MockJwtAdminAuthGuard(user))
+            .overrideProvider(DynamicDataTypeFinderService)
+            .useValue({
+                bootstrapDynamicTypes: async () => {
+                    Promise.resolve();
+                },
+            })
+            .overrideProvider(UnionFactory)
+            .useValue(mockUnionFactory)
+            .overrideProvider(DiscoveryService)
+            .useValue(mockDiscoveryService)
+
+            .compile();
+
+        app = testModule.createNestApplication();
+
+        await app.init();
+
+        memoryMatchRepository = app.get(MEMORY_MATCH_REPOSITORY_INJECTION_TOKEN);
+    };
+
+    beforeEach(async () => {
+        await app
+            .get(ArangoDatabaseProvider)
+            .getDatabaseForCollection('memory_match_rounds')
+            .clear();
+    });
+
+    describe(`when the user is a COSCRAD admin`, () => {
+        const coscradAdminUser = buildTestInstance(CoscradUser, {
+            roles: [CoscradUserRole.superAdmin],
+        });
+
+        beforeAll(async () => {
+            await setItUp(new CoscradUserWithGroups(coscradAdminUser, []));
+        });
+
+        afterAll(async () => {
+            await app.close();
+
+            app.get(ArangoDatabaseProvider).close;
+        });
+
+        describe(`when the request is valid`, () => {
+            beforeEach(async () => {
+                await memoryMatchRepository.create(testRound);
+            });
+
+            it(`should remove the card`, async () => {
+                const endpoint = buildEndpoint(testMemoryRoundId, targetSquenceNumber);
+
+                const res = await request(app.getHttpServer()).delete(endpoint);
+
+                expect(res.status).toBe(HttpStatusCode.ok);
+
+                const updatedRound = (await memoryMatchRepository.fetchById(
+                    testMemoryRoundId
+                )) as MemoryMatchRound;
+
+                expect(updatedRound.count()).toBe(initialSize - 1);
+
+                expect(updatedRound.has(targetSquenceNumber)).toBe(false);
+            });
+        });
+
+        describe(`when the round is not yet published`, () => {
```

Comment:
```ts

describe(`when the round is already published`,()=>{
const publishedRound = buildTestInstance(MemoryMatchRound,{
isPublished: true,
// add a `cardBackImageId` as well
// build MAX_NUMBER_OF_CARD cards and add them- each one must have an audio and image ID
// One of these cards should have the `targetSequenceNumber`
})
)

// double check that you made it publishable
const validationResult = publishedRound.validatePublicationStatus().length 

if(validationResult> 0){
throw new InternalError(`Invalid published round in test setup`,validationResult,)
}
```
You can iterate on running the test until you don't hit the in-line validation error above.

Once you have that, simply attempt to delete the targetSequenceNumber card and check the error.